src/Microsoft.ML.AutoML/Experiment/Experiment.cs

```diff
@@ -49,12 +51,41 @@ internal class Experiment<TRunDetail, TMetrics> where TRunDetail : RunDetail
             _datasetColumnInfo = datasetColumnInfo;
             _runner = runner;
             _logger = logger;
+            _endExperimentWhenAble = false;
+        }
+
+        private void MaxExperimentTimeExpiredEvent(object sender, EventArgs e)
+        {
+            // If at least one model was run, end experiment immediately.
+            // Else, wait for first model to run before experiment is concluded.
+            _endExperimentWhenAble = true;
+            if (_history.Count > 0)
```

Comment:
I think checking for a successful model is slightly more user friendly as the first model may have failed and the second model may be 95% complete. Granted, the difference to the user will be minimal.

Reply:
Good point, thank you!